cli/main.c

```diff
@@ -23,157 +24,55 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
+#include "commands.h"
 #include "glome.h"
 
-const char *kUsage =
+#define GLOME_CLI_MAX_MESSAGE_LENGTH 4095
+
+static const char *kUsage =
     "Usage: \n"
-    "  To generate a new keypair (PRIVATE-KEY will be created if needed)\n"
-    "    %s PRIVATE-KEY >PUBLIC-KEY\n\n"
-    "  To generate a tag (defaults: MESSAGE:='' counter:=0):\n"
-    "    %s PRIVATE-KEY PEER-KEY [MESSAGE [COUNTER]]\n\n"
+    "  To generate a new keypair\n"
+    "    umask 077\n"
+    "    %s genkey | tee PRIVATE-KEY | %s pubkey >PUBLIC-KEY\n\n"
+    "  To generate a tag:\n"
+    "    %s tag --key PRIVATE-KEY --peer PEER-KEY [--counter COUNTER] "
+    "<MESSAGE\n\n"
     "  To verify a tag:\n"
-    "    %s PRIVATE-KEY PEER-KEY MESSAGE COUNTER TAG\n";
-
-static int decode_hex(uint8_t *dst, size_t dst_len, const char *in) {
-  size_t len = strlen(in);
-  if (len > 2 && in[0] == '0' && in[1] == 'x') {
-    len -= 2;
-    in += 2;
-  }
-  if (len > dst_len * 2 || len % 2 != 0) {
-    return -1;
-  }
-  size_t i;
-  for (i = 0; i < len / 2; i++) {
-    if (sscanf(in + (i * 2), "%02hhX", dst + i) != 1) {
-      fprintf(stderr, "ERROR while parsing byte %zu ('%c%c') as hex\n", i,
-              in[2 * i], in[2 * i + 1]);
-      return -3;
-    }
-  }
-  return i;
-}
+    "    %s verify --key PRIVATE-KEY --peer PEER-KEY --tag TAG [--counter "
+    "COUNTER] <MESSAGE\n";
```

Comment:
`<MESSAGE` feels weird to me. `echo MESSAGE |`? Or `<MESSAGE>`?

Reply:
I added `-FILE` suffixes to make the nature of these arguments more obvious. I'd use `<MESSAGE>` in a man page synopsis, but rather not in this example invocation to avoid confusion with shell redirections.